### DISCUSSION TITLE
Custom versioning recipe to version & publish both `dev` and `main`

### DISCUSSION THREAD

**Discussion by sedghi** [lerna/lerna#3705] [Q&A] [answered]
I was wondering what lerna recipe achieve my requirement below. 

I have two branches `dev` and `main` where I occasionally merge `dev` into `main`. I want to version and publish both branches. 
Assume my `main` is at 3.5.0. I would like my `dev ` to be at `3.6.0-beta1` and every time I merge into `dev` it bumps the beta number (so I don't care if it is a minor, or patch, I need it to only bump beta number). Until I merge from `dev` into `main` which makes my `main` 3.6.0 and my `dev` to be `3.7.0.beta1`. 

For versioning and publishing two branches, I haven't find any other versioning scheme to be more readable than this. (I had another thought that maybe, for example, while my main is at 3.5.0, my dev becomes 3.5.1-beta, 3.5.2-beta, 3.5.3-beta but that implies that the 3.5.0 is not stable, but in fact it is). I know this is a weird scenario, and might be a long shot to get an answer here. 

PS: if you have any idea that makes sense to my library and my community (other than the one I mentioned above), I'm open to ditch my custom versioning too

**Comment by fahslaj** [comment]
Hi @sedghi , Lerna would not be able to do all of this in one command. But you could definitely achieve it in CI. After `dev` is merged into `main`, a job could run on `main` that performs a version & publish for 3.6.0. Then, at the end of that job, the next step is to checkout `dev` and perform another version & publish for a prerelease version (like 3.7.0-beta.0). 

If done this way, Lerna can definitely accomplish this, since it is just performing normal version and publish operations, just twice in sequence - once on each branch.

**Reply by sedghi** [reply, marked as answer]
Yeah for anyone interested I did it in CI with node, works great

https://github.com/OHIF/Viewers/blob/master/publish-version.mjs
https://github.com/OHIF/Viewers/blob/master/publish-package.mjs
